http-api-monitor-offers.adoc

```diff
@@ -0,0 +1,327 @@
+
+Here is a full script. You must substitute `EMAIL_ACCOUNT_USERNAME`, `EMAIL_ACCOUNT_PASSWORD`, `EMAIL_FROM_ADDRESS` and `EMAIL_TO_ADDRESS`
+with appropriate values.
+
+[source,javascript]
+----
+const _ = require('lodash');
+const $http = require('http-as-promised');
+const nodemailer = require('nodemailer');
+
+const threshold = 0.1;
+
+const EMAIL_ACCOUNT_USERNAME = 'email@gmail.com';
+const EMAIL_ACCOUNT_PASSWORD = 'secret';
+const EMAIL_FROM_ADDRESS = 'from@gmail.com';
+const EMAIL_TO_ADDRESS = 'to@gmail.com';
+
+function getPriceFilter(marketPrice) {
+    const maxPrice = marketPrice * (1 - threshold) * 10000;
+    return offer => {
+        if (offer.useMarketBasedPrice)
+            return offer.marketPriceMargin >= threshold;
+        return offer.price < maxPrice;
+    }
+}
+
+function getMarketPrice() {
+    return $http.get('http://localhost:8080/api/v1/currencies/prices?currencyCodes=EUR', {resolve: 'body', json: true})
+            .then(body => _.get(body, 'prices.EUR'))
+}
+
+function getOffers() {
+    return $http.get('http://localhost:8080/api/v1/offers', {resolve: 'body', json: true}).then(body => body.offers);
+}
+
+function notify(offers) {
+    if (!offers.length) {
+        console.log('No interesting offers found');
+        return;
+    }
+
+    const transporter = nodemailer.createTransport({
+        service: 'gmail',
+        auth: {
+            user: EMAIL_ACCOUNT_USERNAME,
+            pass: EMAIL_ACCOUNT_PASSWORD
+        }
+    });
+
+    const text = _.map(offers, offer => `${offer.amount / 100000000} BTC (-${_.round(offer.margin * 100, 2)}%)`).join('\n');
+
+    const mailOptions = {
+        from: EMAIL_FROM_ADDRESS,
+        to: EMAIL_TO_ADDRESS,
+        subject: `${offers.length} interesting BTC offers from Bisq`,
+        text: text
+    };
+
+    transporter.sendMail(mailOptions, function (error) {
+        if (error) {
+            console.log(error);
+        } else {
+            console.log(`Notification about ${offers.length} offers sent`);
+        }
+    });
+}
+
+function filterOffers([offers, marketPrice]) {
+    return _(offers)
+            .filter({
+                baseCurrencyCode: 'BTC',
+                counterCurrencyCode: 'EUR',
+                direction: 'SELL',
+                paymentMethodId: 'SEPA'
+            })
+            .filter(i => _.includes(i.acceptedCountryCodes, 'PL'))
+            .filter(getPriceFilter(marketPrice))
+            .map(i => _.pick(i, 'baseCurrencyCode', 'counterCurrencyCode', 'direction', 'paymentMethodId', 'id', 'useMarketBasedPrice', 'price', 'marketPriceMargin', 'amount', 'minAmont'))
+            .map(i => ({amount: i.amount, margin: i.useMarketBasedPrice ? i.marketPriceMargin : marketPrice / i.price}))
+            .value();
+}
+
+Promise.all([getOffers(), getMarketPrice()])
+        .then(filterOffers)
+        .then(notify)
+        .catch(e => console.error(e));
+----
```

Comment:
- [ ] As per the comments above re email, I'd yank all of that and just print to the console.

- [ ] It's probably a good idea to recap the whole file here, I agree, so that the reader can take the whole thing in, see that it (a) makes sense, (b) is short, and (c) can easily copy and paste it if they have failed to copy and paste the individual snippets one-by-one in the steps above.

Reply:
done